packages/ui/src/components/editor/panels/Viewport/container/index.tsx

```diff
@@ -116,6 +158,12 @@ const ViewportDnD = ({ children }: { children: React.ReactNode }) => {
       className={twMerge('h-full w-full border border-white', isDragging ? 'border-4' : 'border-none')}
     >
       {children}
+      {isLoading.value && (
+        // immediately dim the viewport, then show LoadingView after delay
+        <div className="absolute inset-0 z-[80] flex items-center justify-center bg-black bg-opacity-50 transition-opacity duration-300">
+          <LoadingView fullSpace delaySpinner className="block h-12 w-12" />
+        </div>
```

Comment:
Instead of using parent <div> here. You can use containerClass prop of LoadingView to manage container if required.

Also you can remove unused classes as I dont think we would require most of them i.e. flex, etc.